### PR TITLE
Removes apt-get installed packages from Dockerfile

### PR DESCRIPTION
**Pull request by claraberendsen**
### Description
There were some issues with unattended upgrades making builds fail ([See](https://build.osrfoundation.org/job/gz_sim-abichecker-any_to_any-ubuntu-jammy-amd64/120)). After some research it seems this is due to release-tools installing packages that are already provisioned when the image for the agents is built. This PR removes all dependencies that are installed through this mechanism.

### CODE REVIEW

**Comment by Blast545** [comment]
Just for the sake of double checking it doesn't break the build where the problem was found, can you run a job in https://build.osrfoundation.org/job/gz_sim-abichecker-any_to_any-ubuntu-jammy-amd64/build?delay=0sec with this fix branch?

**Review by Blast545 — APPROVED** [review]

**Comment by j-rivero** [comment]
> Just for the sake of double checking it doesn't break the build where the problem was found, can you run a job in https://build.osrfoundation.org/job/gz_sim-abichecker-any_to_any-ubuntu-jammy-amd64/build?delay=0sec with this fix branch?

Clarifying: to avoid the problem of having two conflicting package installation processes we should remove all of the packages in this call to be able to get rid of it. To do so, the provisioning should be patches to install all the packages installed by `NEEDED_HOST_PACKAGES`. I did it in https://github.com/osrf/osrf_jenkins_agent/pull/43.

To completely kill the apt-get install, we would need to merge the PR in provisioning and re-generate new AIMs.

**Review by j-rivero — APPROVED** [review]

**Comment by claraberendsen** [comment]
> To completely kill the apt-get install, we would need to merge the PR in provisioning and re-generate new AIMs.
@j-rivero I'm agree on unifying everything on provisioning. I will update this PR to remove the apt-get call completely and proceed to review and deploy the related one you opened before merging this one.

**Comment by claraberendsen** [comment]
@j-rivero re-requesting a review here since we changed approach.

**Review by j-rivero — APPROVED** [review]

**Comment by j-rivero** [comment]
> @j-rivero re-requesting a review here since we changed approach.

Can not merge before we re-generate the AMIs of the agents.

**Pull request merged**
